pkg/loop/internal/ccip_execution.go

```diff
@@ -286,3 +299,25 @@ func (e *execProviderServer) NewOnRampReader(ctx context.Context, req *ccippb.Ne
 	e.deps.Add(onRampResource)
 	return &ccippb.NewOnRampReaderResponse{OnrampReaderServiceId: int32(onRampID)}, nil
 }
+
+func (e *execProviderServer) NewPriceRegistryReader(ctx context.Context, req *ccippb.NewPriceRegistryReaderRequest) (*ccippb.NewPriceRegistryReaderResponse, error) {
+	reader, err := e.impl.NewPriceRegistryReader(ctx, cciptypes.Address(req.Address))
+	if err != nil {
+		return nil, err
+	}
+	// wrap the reader in a grpc server and serve it
+	priceRegistryHandler := ccipinternal.NewPriceRegistryGRPCServer(reader)
+	// the id is handle to the broker, we will need it on the other side to dial the resource
+	priceReaderID, spawnedServer, err := e.ServeNew("PriceRegistryReader", func(s *grpc.Server) {
+		ccippb.RegisterPriceRegistryReaderServer(s, priceRegistryHandler)
+	})
+	if err != nil {
+		return nil, err
+	}
+	// There is a chicken-and-egg problem here. Our broker is responsible for spawning the grpc server.
```

Comment:
@jmank88 please specific examine this logic and associated test in `price_registry_test.go`